tests/jax/test_interpolatedimage_utils.py

```diff
@@ -0,0 +1,317 @@
+            )
+        )
+        % (10**7),
+        1,
+    )
+
+    rng = np.random.RandomState(seed=seed)
+    if rng.uniform() < 0.75:
+        pytest.skip(
+            "Skipping `test_interpolatedimage_utils_comp_to_galsim` case at random to save time."
+        )
+
+    gimage_in = _galsim.Image(ref_array, scale=0.2)
+    jgimage_in = jax_galsim.Image(ref_array, scale=0.2)
+
+    gii = _galsim.InterpolatedImage(
+        gimage_in,
+        wcs=wcs,
+        offset=_galsim.PositionD(offset_x, offset_y),
+        use_true_center=use_true_center,
+        normalization=normalization,
+        x_interpolant=x_interp,
+    )
+    jgii = jax_galsim.InterpolatedImage(
+        jgimage_in,
+        wcs=jax_galsim.BaseWCS.from_galsim(wcs),
+        offset=jax_galsim.PositionD(offset_x, offset_y),
+        use_true_center=use_true_center,
+        normalization=normalization,
+        x_interpolant=x_interp,
+    )
+
+    np.testing.assert_allclose(gii.stepk, jgii.stepk, rtol=0.5, atol=0)
+    np.testing.assert_allclose(gii.maxk, jgii.maxk, rtol=0.5, atol=0)
+    kxvals = [
+        (0, 0),
+        (-5, -5),
+        (-10, 10),
+        (1, 1),
+        (1, -2),
+        (-1, 0),
+        (0, -1),
+        (-1, -1),
+        (-2, 2),
+        (-5, 0),
+        (3, -4),
+        (-3, 4),
+    ]
+    for x, y in kxvals:
+        if method == "kValue":
+            dk = jgii._original._kim.scale * rng.uniform(low=0.5, high=1.5)
+            np.testing.assert_allclose(
+                gii.kValue(x * dk, y * dk),
+                jgii.kValue(x * dk, y * dk),
+                err_msg=f"kValue mismatch: wcs={wcs}, x={x}, y={y}",
+            )
+        else:
+            dx = jnp.sqrt(jgii._original._wcs.pixelArea()) * rng.uniform(
+                low=0.5, high=1.5
+            )
+            np.testing.assert_allclose(
+                gii.xValue(x * dx, y * dx),
+                jgii.xValue(x * dx, y * dx),
+                err_msg=f"xValue mismatch: wcs={wcs}, x={x}, y={y}",
+            )
+
+
+def _compute_fft_with_numpy_jax_galsim(im):
+    import numpy as np
+
+    from jax_galsim import BoundsI, Image
+
+    No2 = max(-im.bounds.xmin, im.bounds.xmax + 1, -im.bounds.ymin, im.bounds.ymax + 1)
+
+    full_bounds = BoundsI(-No2, No2 - 1, -No2, No2 - 1)
+    if im.bounds == full_bounds:
+        # Then the image is already in the shape we need.
+        ximage = im
+    else:
+        # Then we pad out with zeros
+        ximage = Image(full_bounds, dtype=im.dtype, init_value=0)
+        ximage[im.bounds] = im[im.bounds]
+
+    dx = im.scale
+    # dk = 2pi / (N dk)
+    dk = np.pi / (No2 * dx)
+
+    out = Image(BoundsI(0, No2, -No2, No2 - 1), dtype=np.complex128, scale=dk)
+    out._array = np.fft.fftshift(np.fft.rfft2(np.fft.fftshift(ximage.array)), axes=0)
+    out *= dx * dx
+    out.setOrigin(0, -No2)
+    return out
+
+
+@pytest.mark.parametrize("n", [5, 4])
+def test_interpolatedimage_utils_jax_galsim_fft_vs_galsim_fft(n):
```

Comment:
in this test, it doesn't seem like you are using `galsim` just `jax_galsim`, was that intentional?

Reply:
This is an old test name. I removed the galsim comps since they were redunant with the galsim test suite. I have change the test name.